plugins/quay/src/api/index.ts

```diff
@@ -60,8 +65,12 @@ export class QuayApiClient implements QuayApiV1 {
   }
 
   private async fetcher(url: string) {
+    const {token: idToken} = await this.identityApi.getCredentials();
     const response = await fetch(url, {
-      headers: { 'Content-Type': 'application/json' },
+      headers: {
+        'Content-Type': 'application/json',
+        ...( idToken && {'Authorization': `Bearer ${idToken}`})
```

Comment:
is this needed, as with proxy header are propagated automatically and won't be seen on the client side. This plugin I have verified and similar should be case with Jfrog as well https://backstage.io/docs/tutorials/using-backstage-proxy-within-plugin/

Reply:
@pbuehl-redhat it seems to me like you're trying to implement https://backstage.io/docs/auth/service-to-service-auth#usage-in-backend-plugins for plugins which **don't** have a backend plugin counterpart.

Reply:
@invincibleJai I think I'm finding the answers myself. You can run the proxy with auth middleware. In that case you need to be able to authenticate against it.

Reply:
However, as I've learnt now, you can make Proxy to use backend service auth as well. In that case this applies to the  plugins which use proxy as well.

```ts
  apiRouter.use('/proxy', authMiddleware, await proxy(proxyEnv));
```